src/warc2zim/content_rewriting/html.py

```diff
@@ -195,81 +215,412 @@ def handle_pi(self, data: str):
+    ):
+        return (attr_name, "text/html; charset=UTF-8")
+
+
+@rules.rewrite_attribute()
+def rewrite_onxxx_tags(
+    attr_name: str, attr_value: str | None, js_rewriter: JsRewriter
+) -> AttrNameAndValue | None:
+    """Rewrite onxxx script attributes"""
+    if attr_value and attr_name.startswith("on") and not attr_name.startswith("on-"):
+        return (attr_name, js_rewriter.rewrite(attr_value))
+
+
+@rules.rewrite_attribute()
+def rewrite_style_tags(
+    attr_name: str, attr_value: str | None, css_rewriter: CssRewriter
+) -> AttrNameAndValue | None:
+    """Rewrite style attributes"""
+    if attr_value and attr_name == "style":
+        return (attr_name, css_rewriter.rewrite_inline(attr_value))
+
+
+@rules.rewrite_attribute()
+def rewrite_href_src_attributes(
+    tag: str,
+    attr_name: str,
+    attr_value: str | None,
+    attrs: AttrsList,
+    url_rewriter: ArticleUrlRewriter,
+    base_href: str | None,
+    notify_js_module: Callable[[ZimPath], None],
+):
+    """Rewrite href and src attributes
+
+    This is also notifying of any JS script found used as a module, so that this script
+    is properly rewritten when encountered later on.
+    """
+    if attr_name not in ("href", "src") or not attr_value:
+        return
+    if get_html_rewrite_context(tag=tag, attrs=attrs) == "js-module":
+        notify_js_module(url_rewriter.get_item_path(attr_value, base_href=base_href))
+    return (
+        attr_name,
+        url_rewriter(attr_value, base_href=base_href, rewrite_all_url=tag != "a"),
+    )
+
+
+@rules.rewrite_attribute()
+def rewrite_srcset_attribute(
+    attr_name: str,
+    attr_value: str | None,
+    url_rewriter: ArticleUrlRewriter,
+    base_href: str | None,
+):
+    """Rewrite srcset attributes"""
+    if attr_name != "srcset" or not attr_value:
+        return
+    value_list = attr_value.split(",")
+    new_value_list = []
+    for value in value_list:
+        url, *other = value.strip().split(" ", maxsplit=1)
+        new_url = url_rewriter(url, base_href=base_href)
+        new_value = " ".join([new_url, *other])
+        new_value_list.append(new_value)
+    return (attr_name, ", ".join(new_value_list))
+
+
+@rules.rewrite_tag()
+def rewrite_base_tag(tag: str, attrs: AttrsList, *, auto_close: bool):
+    """Handle special case of <base> tag which have to be simplified (remove href)
+
+    This is special because resulting tag might be empty and hence needs to be dropped
+    """
+    if tag != "base":
+        return
+    if extract_attr(attrs, "href") is None:
+        return  # needed so that other rules will be processed as well
+    values = " ".join(
+        format_attr(*attr)
+        for attr in [
+            (attr_name, attr_value)
+            for (attr_name, attr_value) in attrs
+            if attr_name != "href"
+        ]
+    )
+    if values:
+        return f"<base {values}{'/>' if auto_close else '>'}"
+    else:
+        return ""  # drop whole tag
```

Comment:
Hum, seems a bit fragile but we can live with it

Reply:
Yep, especially since it is heavily tested